network-config-analyzer/ConnectionSet.py

```diff
@@ -26,11 +26,59 @@ def __eq__(self, other):
             return self.allow_all == other.allow_all and self.allowed_protocols == other.allowed_protocols
         return NotImplemented
 
+    def __lt__(self, other):
+        if self.allow_all:
+            return False
+        if other.allow_all:
+            return True
+        return len(self.allowed_protocols) < len(other.allowed_protocols)
+
+    def __hash__(self):
+        return hash((frozenset(self.allowed_protocols.keys()), self.allow_all))
+
+    def get_connections_list(self):
+        res = []
+        if self.allow_all:
+            res.append(str(self))
+            return res
+        if not self.allowed_protocols:
+            res.append(str(self))
+            return res
+        for protocol in [6, 17, 132]:
```

Comment:
updated implementation to support all protocols + added the delegation for properties to their classes. Note that for semantic-diff there may be a situation of "all connections" - "some protocol (e.g. TCP)", and the result here may depend on whether the policy is in k8s or calico (all connections of k8s is different from all connections of calico, since only the protocols that I specified originally are supported in k8s). I added a flag to identify which policy type should be considered.

Reply:
Good catch